🎉 Improve week numbers

albinekb opened the pull request:
Fixes the aligmet issue: https://github.com/wix/react-native-calendars/pull/239#issuecomment-349905968

- Adds prop types for showWeekNumber
- Documents showWeekNumber
- Changes to using the `<Day />` component for week numbers
- Shows correct number for week number 53 (due to leap years)

I will do another PR that adds a prop to override the styles, component and onPress for the week number 👍

albinekb commented:
@tautvilas 👀  https://github.com/wix/react-native-calendars/pull/239

tautvilas commented:
Hey, I mean not only alignment but also logic, why 2nd week of Jan is 53, but not 2?

albinekb commented:
Oh, I didn't notice that. Fixing 🚧

albinekb commented:
![image](https://user-images.githubusercontent.com/5027156/33718665-84dd0f84-db5e-11e7-96a5-6a89388255f9.png)
@tautvilas Looks like it's a bug in xdate? I can open an issue over there, we got 2 years on us to fix it 👍

albinekb commented:
Anything I can do to help this move a bit quicker?  🙏

albinekb commented:
@tautvilas ping :shipit:

tautvilas commented:
Hi, if this functionality is broken in XDate I don't see how we can implement this properly soon. The options I see now:

1) PR to XDate
2) Implement our own logic to calculate week number (you could create a PR for calendars)
3) Migrate librare from XDate to momentJS (probably not gonna happen soon)

Alternatively I can only propose for you to create a fork that matches your needs..

albinekb commented:
It's only a problem in 2021, but ok if you want to fix it before that. Would you accept a PR that replaces `xdate` with `date-fns` or `moment.js`?

albinekb commented:
@tautvilas here's the issue in xdate: https://github.com/arshaw/xdate/issues/37
I don't know what's correct, xterm or moment.js, but it has nothing to do with this PR really. As we already merged the weekNumber functionality, this only improves on that and cleans it up. 

I will investigate the incorrect weekNumber in the first week `2021` and send a PR that corrects it.

albinekb commented:
Bug seen from last PR ![image](https://user-images.githubusercontent.com/5027156/34128942-9e8cb23a-e442-11e7-8539-9ea916409f09.png)

Fixed with getting the last day in the week: ![image](https://user-images.githubusercontent.com/5027156/34128964-b3e42f82-e442-11e7-8b5a-74897efec9f5.png)

albinekb commented:
Sorry for the confusion @tautvilas. 

TL:DR: the week number issue that you wanted me to fix in the last PR is now fixed. I've checked the years between too and it looks correct to me (ends with week 52, starts with week 1)

Bug seen in last PR
![image](https://user-images.githubusercontent.com/5027156/34128942-9e8cb23a-e442-11e7-8539-9ea916409f09.png)

Fixed with getting the last day in the week:
![image](https://user-images.githubusercontent.com/5027156/34128964-b3e42f82-e442-11e7-8b5a-74897efec9f5.png)

💥 We don't need to change anything in xdate or replace it with moment. Getting the last day in the week always yields the correct week number.

tautvilas commented:
Thx, will review this thursday

tautvilas commented:
Now it looks good:) published with react-native-calendars@1.16.0

albinekb commented:
Awesome!  Sorry for the confusion. Next week I'll PR some improvements to make it customizable like the other options. Thanks 🙏 🎉

jarrx commented:
It seems like week numbers does not work with the CalendarList component.

albinekb commented:
@jarrx that's correct. I only implemented it for the calendar component, I can see if I can get it into CalendarList easy.

jarrx commented:
@albinekb that would be awsome! According to the docs, CalendarList should be able to take all props that the Calendar component do and since CalendarList is more or less a composition of multiple calendar components maybe it’s not too hard to implement. Otherwise great job!

jarrx commented:
@albinekb Any progress on this?

albinekb commented:
Just got back to work, I will mosy likely send a PR mid-week 👍 :pray:

jarrx commented:
@albinekb I understand that you didn't had the time to fix this, but do you think you'll send a PR anytime in the future?

albinekb commented:
Yes, I actually got this in my sprint next week @jarrx as we changed over to the `CalendarList` in our app, so it will happen next week. Sorry I didn't get around to it!

jarrx commented:
@albinekb I like your job with the week numbers but are you able to custom style them?
I can't find out how to change the styling other than passing 'textDisabledColor' to theme, which also changes the color of e.g. extra days.
Feels like the week numbers are too similar to the other days.
